### ISSUE TITLE
Diamond no longer compiles on aarch64/arm64 and other archs as of 2.1.0 

### ISSUE THREAD

**Issue by mr-c** [bbuchfink/diamond#655]
Hello,

Thanks for the new release. Alas it breaks support for non-x86 archs which is affecting Debian and BioConda

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1030360
https://github.com/bioconda/bioconda-recipes/pull/39008

Please consider the use of SIMD Everywhere to enable compilation on non-x86 systems with little effort on your side. It would compliment hand-written support like https://github.com/bbuchfink/diamond/pull/615 

https://github.com/simd-everywhere/simde

**Comment by bbuchfink** [comment]
Mea culpa, the release was rushed and I forgot to take care of this. Will fix.

**Comment by tillea** [comment]
Please note:  Fixing this issue is urgent in case you want to see the latest version of diamond inside the next stable release.

**Comment by bbuchfink** [comment]
@tillea How urgent, does tomorrow work for you?

**Comment by bbuchfink** [comment]
I hope I caught all the errors in the latest release.

**Comment by mr-c** [comment]
Thank you @bbuchfink ; I'm testing the 2.1.1 release now

**Comment by mr-c** [comment]
With 2.1.1 I get a segfault on arm64 while running tests; I'll try 2.1.2 now
https://buildd.debian.org/status/fetch.php?pkg=diamond-aligner&arch=arm64&ver=2.1.1-1&stamp=1676897247&raw=0

**Comment by bbuchfink** [comment]
Ok, trying to setup a CI for arm64.

**Comment by mr-c** [comment]
Thanks, let me know if you want assistance. I have experience using a variety of CI providers to do  native build + test, and cross-compiling and running via qemu 

2.1.2 also segfaulted in qemu; I can test on real hardware if you'd like

**Comment by bbuchfink** [comment]
I'm getting the same segfault in cirrus CI, but only for the release build, which makes it a bit more difficult. I know how to cross compile, but if you could let me know how to best test this using qemu on a ubuntu/x86 system, that would be very helpful.

**Comment by bbuchfink** [comment]
Update: I have managed to boot up an arm64 ubuntu image using qemu, but not managed to get internet access within the VM so far.

**Comment by mr-c** [comment]
@bbuchfink I can get you access to the Debian arm64 porting box ; I think you can start the process at https://nm.debian.org/wizard/process/ga/  ; you'll want to request access to [`amdahl`](https://db.debian.org/machines.cgi?host=amdahl) ; and my Debian username is `crusoe`

**Comment by bbuchfink** [comment]
I'm not able to register for the salsa. It says I failed to solve the captcha but there's no captcha showing on the page. Signing in with google does not work either, it says it is limited to the organization.

**Comment by mr-c** [comment]
@bbuchfink Sorry, I guess it is undergoing maintenance right now https://lists.debian.org/debian-infrastructure-announce/2023/02/msg00000.html

**Comment by mr-c** [comment]
@bbuchfink Salsa maintenance is finished. Can you try again?

**Comment by bbuchfink** [comment]
I already worked on it using an arm64 aws instance. Fixed it using a strategically placed `__attribute__ ((noinline))`: https://github.com/bbuchfink/diamond/commit/cd02714ac51b9be45451098fdf6df3b739248742

Turns out the issue is specific to gcc 12 (and arm64). Without the noinline, apparently the `BuildCallback::finish` function in the same file is simply not called. Looks like a compiler bug if you ask me.
